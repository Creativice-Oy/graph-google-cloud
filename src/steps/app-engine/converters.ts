import {
  createIntegrationEntity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import { appengine_v1 } from 'googleapis';
import { getGoogleCloudConsoleWebLink } from '../../utils/url';
import {
  ENTITY_CLASS_APP_ENGINE_APPLICATION,
  ENTITY_CLASS_APP_ENGINE_INSTANCE,
  ENTITY_CLASS_APP_ENGINE_SERVICE,
  ENTITY_CLASS_APP_ENGINE_VERSION,
  ENTITY_TYPE_APP_ENGINE_APPLICATION,
  ENTITY_TYPE_APP_ENGINE_INSTANCE,
  ENTITY_TYPE_APP_ENGINE_SERVICE,
  ENTITY_TYPE_APP_ENGINE_VERSION,
} from './constants';

export function createAppEngineApplicationEntity(
  data: appengine_v1.Schema$Application,
  projectId: string,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_APP_ENGINE_APPLICATION,
        _type: ENTITY_TYPE_APP_ENGINE_APPLICATION,
        _key: data.name as string,
        name: data.name,
        displayName: data.name as string,
        location: data.locationId,
        servingStatus: data.servingStatus,
        defaultHostname: data.defaultHostname,
        defaultBucket: data.defaultBucket,
        splitHealthChecks: data.featureSettings?.splitHealthChecks,
        useContainerOptimizedOs: data.featureSettings?.useContainerOptimizedOs,
        webLink: getGoogleCloudConsoleWebLink(
          `/appengine?project=${projectId}`,
        ),
      },
    },
  });
}

export function createAppEngineServiceEntity(
  data: appengine_v1.Schema$Service,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_APP_ENGINE_SERVICE,
        _type: ENTITY_TYPE_APP_ENGINE_SERVICE,
        _key: data.name as string,
        name: data.name,
        displayName: data.name as string,
      },
    },
  });
}

export function createAppEngineVersionEntity(
  data: appengine_v1.Schema$Version,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_APP_ENGINE_VERSION,
        _type: ENTITY_TYPE_APP_ENGINE_VERSION,
        _key: data.name as string,
        name: data.name,
        displayName: data.name as string,
        versionUrl: data.versionUrl,
        category: ['application'],
        createdOn: parseTimePropertyValue(data.createTime),
      },
    },
  });
}

export function createAppEngineInstanceEntity(
  data: appengine_v1.Schema$Instance,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_APP_ENGINE_INSTANCE,
        _type: ENTITY_TYPE_APP_ENGINE_INSTANCE,
        _key: data.name as string,
        name: data.name,
        displayName: data.name as string,
        availability: data.availability,
        requests: data.requests,
        createdOn: parseTimePropertyValue(data.startTime),
      },
    },
  });
}
