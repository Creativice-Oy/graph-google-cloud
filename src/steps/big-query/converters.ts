import { bigquery_v2 } from 'googleapis';
import { createGoogleCloudIntegrationEntity } from '../../utils/entity';
import {
  BIG_QUERY_DATASET_ENTITY_CLASS,
  BIG_QUERY_DATASET_ENTITY_TYPE,
  BIG_QUERY_TABLE_ENTITY_CLASS,
  BIG_QUERY_TABLE_ENTITY_TYPE,
} from './constants';
import { isMemberPublic } from '../../utils/iam';
import { BigQueryTable } from './client';
import { getGoogleCloudConsoleWebLink } from '../../utils/url';

interface DatasetAccess {
  domain?: string;
  groupByEmail?: string;
  iamMember?: string;
  role?: string;
  specialGroup?: string;
  userByEmail?: string;
}

// 7.1 Ensure that BigQuery datasets are not anonymously or publicly accessible (Scored)
function isBigQueryDatasetPublicAccess(accessList: DatasetAccess[]): boolean {
  for (const access of accessList) {
    const value =
      access.groupByEmail ||
      access.userByEmail ||
      access.iamMember ||
      access.specialGroup ||
      '';
    if (isMemberPublic(value)) {
      return true;
    }
  }

  return false;
}

export function createBigQueryDatasetEntity(data: bigquery_v2.Schema$Dataset) {
  return createGoogleCloudIntegrationEntity(data, {
    entityData: {
      source: data,
      assign: {
        _key: data.id as string,
        _type: BIG_QUERY_DATASET_ENTITY_TYPE,
        _class: BIG_QUERY_DATASET_ENTITY_CLASS,
        name: data.datasetReference?.datasetId,
        description: data.description,
        public: isBigQueryDatasetPublicAccess(data.access || []),
        kmsKeyName: data.defaultEncryptionConfiguration?.kmsKeyName,
        location: data.location,
        encrypted: true,
        classification: null,
        etag: data.etag,
        createdOn: data.creationTime
          ? parseInt(data.creationTime, 10)
          : undefined,
        modifiedOn: data.lastModifiedTime
          ? parseInt(data.lastModifiedTime, 10)
          : undefined,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigquery?project=${data.datasetReference?.projectId}&d=${data.datasetReference?.datasetId}&p=${data.datasetReference?.projectId}&page=dataset`,
        ),
      },
    },
  });
}

export function createBigQueryTableEntity({
  data,
  projectId,
  isPublic,
}: {
  data: BigQueryTable;
  projectId: string;
  isPublic: boolean;
}) {
  return createGoogleCloudIntegrationEntity(data, {
    entityData: {
      source: data,
      assign: {
        _key: data.id as string,
        _type: BIG_QUERY_TABLE_ENTITY_TYPE,
        _class: BIG_QUERY_TABLE_ENTITY_CLASS,
        name: data.tableReference?.tableId,
        public: isPublic,
        type: data.type,
        friendlyName: data.friendlyName,
        classification: null,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigquery?project=${projectId}&d=${data.tableReference?.datasetId}&p=${projectId}&t=${data.tableReference?.tableId}&page=table`,
        ),
        createdOn: data.creationTime
          ? parseInt(data.creationTime, 10)
          : undefined,
        expirationTime: data.expirationTime
          ? parseInt(data.expirationTime, 10)
          : undefined,
      },
    },
  });
}
