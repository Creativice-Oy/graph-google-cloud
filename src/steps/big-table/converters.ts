// import { Entity } from '@jupiterone/integration-sdk-core';
import { bigtableadmin_v2 } from 'googleapis';
import { createGoogleCloudIntegrationEntity } from '../../utils/entity';
import { getGoogleCloudConsoleWebLink } from '../../utils/url';
import {
  ENTITY_CLASS_BIG_TABLE_APP_PROFILE,
  ENTITY_CLASS_BIG_TABLE_BACKUP,
  ENTITY_CLASS_BIG_TABLE_CLUSTER,
  ENTITY_CLASS_BIG_TABLE_INSTANCE,
  ENTITY_CLASS_BIG_TABLE_LOCATION,
  ENTITY_CLASS_BIG_TABLE_OPERATION,
  ENTITY_CLASS_BIG_TABLE_TABLE,
  ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
  ENTITY_TYPE_BIG_TABLE_BACKUP,
  ENTITY_TYPE_BIG_TABLE_CLUSTER,
  ENTITY_TYPE_BIG_TABLE_INSTANCE,
  ENTITY_TYPE_BIG_TABLE_LOCATION,
  ENTITY_TYPE_BIG_TABLE_OPERATION,
  ENTITY_TYPE_BIG_TABLE_TABLE,
} from './constants';

export function getOperationKey(operation: bigtableadmin_v2.Schema$Operation) {
  return `bigtable_operation:${operation.name}`;
}

export function getInstanceKey(instance: bigtableadmin_v2.Schema$Instance) {
  return `bigtable_instance:${instance.name}`;
}

export function getAppProfileKey(
  appProfile: bigtableadmin_v2.Schema$AppProfile,
) {
  return `bigtable_appProfile:${appProfile.name}`;
}

export function getClusterKey(cluster: bigtableadmin_v2.Schema$Cluster) {
  return `bigtable_cluster:${cluster.name}`;
}

export function getBackupKey(backup: bigtableadmin_v2.Schema$Backup) {
  return `bigtable_backup:${backup.name}`;
}

export function getTableKey(table: bigtableadmin_v2.Schema$Table) {
  return `bigtable_table:${table.name}`;
}

export function getLocationKey(location: bigtableadmin_v2.Schema$Location) {
  return `bigtable_location:${location.locationId}`;
}

export function createOperationEntity({
  operation,
}: {
  operation: bigtableadmin_v2.Schema$Operation;
}) {
  return createGoogleCloudIntegrationEntity(operation, {
    entityData: {
      source: operation,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_OPERATION,
        _type: ENTITY_TYPE_BIG_TABLE_OPERATION,
        _key: getOperationKey(operation),
        name: operation.name,
        done: operation.done,
      },
    },
  });
}

export function createInstanceEntity({
  instance,
  projectId,
}: {
  instance: bigtableadmin_v2.Schema$Instance;
  projectId: string | undefined | null;
}) {
  return createGoogleCloudIntegrationEntity(instance, {
    entityData: {
      source: instance,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_INSTANCE,
        _type: ENTITY_TYPE_BIG_TABLE_INSTANCE,
        _key: getInstanceKey(instance),
        name: instance.name,
        displayName: instance.displayName!,
        state: instance.state,
        type: instance.type,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigtable/instances/${instance.name}/overview?project=${projectId}`,
        ),
      },
    },
  });
}

export function createAppProfileEntity({
  appProfile,
  projectId,
  instanceId,
}: {
  appProfile: bigtableadmin_v2.Schema$AppProfile;
  projectId: string | undefined | null;
  instanceId: string | undefined | null;
}) {
  return createGoogleCloudIntegrationEntity(appProfile, {
    entityData: {
      source: appProfile,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_APP_PROFILE,
        _type: ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
        _key: getAppProfileKey(appProfile),
        name: appProfile.name,
        instanceId,
        etag: appProfile.etag,
        description: appProfile.description,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigtable/instances/${instanceId}/app-profiles/${appProfile.name}?project=${projectId}`,
        ),
      },
    },
  });
}

export function createClusterEntity({
  cluster,
  projectId,
  instanceId,
}: {
  cluster: bigtableadmin_v2.Schema$Cluster;
  projectId: string | undefined | null;
  instanceId: string | undefined | null;
}) {
  return createGoogleCloudIntegrationEntity(cluster, {
    entityData: {
      source: cluster,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_CLUSTER,
        _type: ENTITY_TYPE_BIG_TABLE_CLUSTER,
        _key: getClusterKey(cluster),
        name: cluster.name,
        instanceId,
        state: cluster.state,
        serveNodes: cluster.serveNodes,
        defaultStorageType: cluster.defaultStorageType,
        location: cluster.location,
        kmsKeyName: cluster.encryptionConfig?.kmsKeyName,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigtable/instances/${instanceId}/overview?project=${projectId}`,
        ),
      },
    },
  });
}

export function createBackupEntity({
  backup,
  projectId,
  instanceId,
  clusterId,
}: {
  backup: bigtableadmin_v2.Schema$Backup;
  projectId: string | undefined | null;
  instanceId: string | undefined | null;
  clusterId: string | undefined | null;
}) {
  return createGoogleCloudIntegrationEntity(backup, {
    entityData: {
      source: backup,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_BACKUP,
        _type: ENTITY_TYPE_BIG_TABLE_BACKUP,
        _key: getBackupKey(backup),
        name: backup.name,
        instanceId,
        clusterId,
        sourceTable: backup.sourceTable,
        expireTime: backup.expireTime,
        startTime: backup.startTime,
        endTime: backup.endTime,
        sizeBytes: backup.sizeBytes,
        state: backup.state,
        encryptionType: backup.encryptionInfo?.encryptionType,
        kmsKeyVersion: backup.encryptionInfo?.kmsKeyVersion,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigtable/instances/${instanceId}/backups?project=${projectId}`,
        ),
      },
    },
  });
}

export function createTableEntity({
  table,
  projectId,
  instanceId,
}: {
  table: bigtableadmin_v2.Schema$Table;
  projectId: string | undefined | null;
  instanceId: string | undefined | null;
}) {
  return createGoogleCloudIntegrationEntity(table, {
    entityData: {
      source: table,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_TABLE,
        _type: ENTITY_TYPE_BIG_TABLE_TABLE,
        _key: getTableKey(table),
        name: table.name,
        classification: 'unclassified', // not sure what to put here. this is a required field for the class DataCollection
        instanceId,
        granularity: table.granularity || undefined, // these value are there in the type definition in the docs, but are not being sent by the API. we're only getting the name
        sourceType: table.restoreInfo?.sourceType || undefined,
        backup: table.restoreInfo?.backupInfo?.backup || undefined,
        webLink: getGoogleCloudConsoleWebLink(
          `/bigtable/instances/${instanceId}/tables?project=${projectId}`,
        ),
      },
    },
  });
}

export function createLocationEntity({
  location,
}: {
  location: bigtableadmin_v2.Schema$Location;
}) {
  return createGoogleCloudIntegrationEntity(location, {
    entityData: {
      source: location,
      assign: {
        _class: ENTITY_CLASS_BIG_TABLE_LOCATION,
        _type: ENTITY_TYPE_BIG_TABLE_LOCATION,
        _key: getLocationKey(location),
        name: location.name,
        locationId: location.locationId,
        displayName: location.displayName || undefined,
      },
    },
  });
}
