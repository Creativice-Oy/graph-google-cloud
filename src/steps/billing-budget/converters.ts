import { billingbudgets_v1 } from 'googleapis';
import { createGoogleCloudIntegrationEntity } from '../../utils/entity';
import {
  ENTITY_CLASS_BILLING_BUDGET,
  ENTITY_TYPE_BILLING_BUDGET,
} from './constants';

export function createBillingBudgetEntity(
  data: billingbudgets_v1.Schema$GoogleCloudBillingBudgetsV1Budget,
) {
  return createGoogleCloudIntegrationEntity(data, {
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_BILLING_BUDGET,
        _type: ENTITY_TYPE_BILLING_BUDGET,
        _key: data.name as string,
        name: data.name,
        displayName: data.displayName as string,
      },
    },
  });
}
