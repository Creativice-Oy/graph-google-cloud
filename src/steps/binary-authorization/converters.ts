import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { binaryauthorization_v1 } from 'googleapis';
import { getGoogleCloudConsoleWebLink } from '../../utils/url';
import {
  BINARY_AUTHORIZATION_POLICY_ENTITY_CLASS,
  BINARY_AUTHORIZATION_POLICY_ENTITY_TYPE,
} from './constants';

export function createBinaryAuthorizationPolicyEntity(
  data: binaryauthorization_v1.Schema$Policy,
  projectId: string,
) {
  const admissionWhitelistPatternsSrc: binaryauthorization_v1.Schema$AdmissionWhitelistPattern[] =
    data.admissionWhitelistPatterns || [];

  const admissionWhitelistPatterns: string[] = admissionWhitelistPatternsSrc
    .map(
      (pattern: binaryauthorization_v1.Schema$AdmissionWhitelistPattern) =>
        pattern.namePattern || '',
    )
    .filter((value) => value);

  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _key: data.name as string,
        _type: BINARY_AUTHORIZATION_POLICY_ENTITY_TYPE,
        _class: BINARY_AUTHORIZATION_POLICY_ENTITY_CLASS,
        name: data.name,
        // 6.1.4 Minimize Container Registries to only those approved (Not Scored)
        admissionWhitelistPatterns: admissionWhitelistPatterns,
        evaluationMode: data.defaultAdmissionRule?.evaluationMode,
        globalPolicyEvaluationMode: data.globalPolicyEvaluationMode,
        title: 'Binary Authorization Policy',
        summary:
          'Binary Authorization implements a policy model, where a policy is a set of rules that governs the deployment of container images to a GKE cluster. Rules in a policy provide specific criteria that an image must satisfy before it can be deployed.',
        content: '',
        webLink: getGoogleCloudConsoleWebLink(
          `/security/binary-authorization/policy?project=${projectId}&folder=&organizationId=`,
        ),
      },
    },
  });
}
