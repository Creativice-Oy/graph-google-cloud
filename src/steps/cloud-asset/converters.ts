import { cloudasset_v1 } from 'googleapis';
import { snakeCase } from 'lodash';
import { hashArray } from '../../utils/crypto';

import { createGoogleCloudIntegrationEntity } from '../../utils/entity';
import { bindingEntities } from './constants';

export function buildIamBindingEntityKey({
  binding,
  projectName,
  resource,
}: {
  binding: cloudasset_v1.Schema$Binding;
  projectName: string | undefined | null;
  resource: string | undefined | null;
}) {
  // There are no unique values returned from this result so use everything
  // available to make the binding key unique.
  const keyBuilders = ['binding'];

  if (projectName) keyBuilders.push('project:' + projectName);
  if (resource) keyBuilders.push('resource:' + resource);
  if (binding.role) keyBuilders.push('role:' + binding.role);
  if (binding.members)
    keyBuilders.push('members:' + hashArray(binding.members));

  return keyBuilders.join('|');
}

export function createIamBindingEntity({
  _key,
  projectId,
  binding,
  resource,
}: {
  _key: string;
  projectId?: string;
  binding: cloudasset_v1.Schema$Binding;
  resource: string | undefined | null;
}) {
  const namePrefix = 'Role Binding for Resource: ';

  return createGoogleCloudIntegrationEntity(binding, {
    entityData: {
      source: binding,
      assign: {
        _class: bindingEntities.BINDINGS._class,
        _type: bindingEntities.BINDINGS._type,
        _key,
        name: snakeCase(namePrefix) + resource,
        displayName: namePrefix + resource,
        resource,
        role: binding.role,
        members: binding.members,
        projectId,
        'condition.title': binding.condition?.title,
        'condition.description': binding.condition?.description,
        'condition.expression': binding.condition?.expression,
      },
    },
  });
}
