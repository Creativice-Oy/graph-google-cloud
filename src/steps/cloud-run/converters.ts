import {
  createIntegrationEntity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import { run_v1 } from 'googleapis';
import { getGoogleCloudConsoleWebLink } from '../../utils/url';
import {
  ENTITY_CLASS_CLOUD_RUN_SERVICE,
  ENTITY_TYPE_CLOUD_RUN_SERVICE,
  ENTITY_TYPE_CLOUD_RUN_ROUTE,
  ENTITY_CLASS_CLOUD_RUN_ROUTE,
  ENTITY_CLASS_CLOUD_RUN_CONFIGURATION,
  ENTITY_TYPE_CLOUD_RUN_CONFIGURATION,
} from './constants';

export function createCloudRunServiceEntity(
  data: run_v1.Schema$Service,
  projectId: string,
) {
  // Build webLink
  let webLink = '';
  if (
    data.metadata?.labels &&
    data.metadata?.labels['cloud.googleapis.com/location']
  ) {
    webLink = `/run/detail/${data.metadata?.labels['cloud.googleapis.com/location']}/${data.metadata.name}`;
  } else {
    // Worst case we lead the user to "view all services" page
    webLink = `/run?project=${projectId}`;
  }

  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_CLOUD_RUN_SERVICE,
        _type: ENTITY_TYPE_CLOUD_RUN_SERVICE,
        _key: data.metadata?.selfLink as string,
        name: data.metadata?.name,
        displayName: data.metadata?.name as string,
        category: ['infrastructure'],
        createdOn: parseTimePropertyValue(data.metadata?.creationTimestamp),
        webLink: getGoogleCloudConsoleWebLink(webLink),
      },
    },
  });
}

export function createCloudRunRouteEntity(data: run_v1.Schema$Route) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_CLOUD_RUN_ROUTE,
        _type: ENTITY_TYPE_CLOUD_RUN_ROUTE,
        _key: data.metadata?.selfLink as string,
        name: data.metadata?.name,
        displayName: data.metadata?.name as string,
        url: data.status?.url,
        createdOn: parseTimePropertyValue(data.metadata?.creationTimestamp),
      },
    },
  });
}

export function createCloudRunConfigurationEntity(
  data: run_v1.Schema$Configuration,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: ENTITY_CLASS_CLOUD_RUN_CONFIGURATION,
        _type: ENTITY_TYPE_CLOUD_RUN_CONFIGURATION,
        _key: data.metadata?.selfLink as string,
        name: data.metadata?.name,
        displayName: data.metadata?.name as string,
        createdOn: parseTimePropertyValue(data.metadata?.creationTimestamp),
      },
    },
  });
}
