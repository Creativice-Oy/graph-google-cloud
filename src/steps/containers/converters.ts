import { container_v1 } from 'googleapis';
import {
  createIntegrationEntity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import {
  CONTAINER_CLUSTER_ENTITY_CLASS,
  CONTAINER_CLUSTER_ENTITY_TYPE,
} from './constants';

export function createContainerClusterEntity(
  data: container_v1.Schema$Cluster,
) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _key: data.selfLink as string,
        _type: CONTAINER_CLUSTER_ENTITY_TYPE,
        _class: CONTAINER_CLUSTER_ENTITY_CLASS,
        createdOn: parseTimePropertyValue(data.createTime),
        name: data.name,
        description: data.description,
        initialNodeCode: data.initialNodeCount,
        nodeMachineType: data.nodeConfig?.machineType,
        nodeDiskSizeGb: data.nodeConfig?.diskSizeGb,
        nodeOAuthScopes: data.nodeConfig?.oauthScopes,
        nodeImageType: data.nodeConfig?.imageType,
        nodePreemptible: data.nodeConfig?.preemptible,
        nodeDiskType: data.nodeConfig?.diskType,
        nodeEnableIntegrityMonitoring:
          data.nodeConfig?.shieldedInstanceConfig?.enableIntegrityMonitoring,
        nodeEnableSecureBoot:
          data.nodeConfig?.shieldedInstanceConfig?.enableSecureBoot,
        network: data.networkConfig?.network,
        clusterIpv4Cidr: data.clusterIpv4Cidr,
        kubernetesDashboardEnabled:
          data.addonsConfig?.kubernetesDashboard?.disabled !== true,
        networkPolicyConfigEnabled:
          data.addonsConfig?.networkPolicyConfig?.disabled !== true,
        subnetwork: data.networkConfig?.subnetwork,
        locations: data.locations,
        useRoutes: data.ipAllocationPolicy?.useRoutes === true,
        useIpAliases: data.ipAllocationPolicy?.useIpAliases === true,
        databaseEncryptionState: data.databaseEncryption?.state,
        databaseEncryptionKey: data.databaseEncryption?.keyName,
        releaseChannel: data.releaseChannel?.channel,
        endpoint: data.endpoint,
        initialClusterVersion: data.initialClusterVersion,
        currentMasterVersion: data.currentMasterVersion,
        status: data.status,
        nodeIpv4CidrSize: data.nodeIpv4CidrSize,
        servicesIpv4Cidr: data.servicesIpv4Cidr,
        instanceGroupUrls: data.instanceGroupUrls,
        currentNodeCount: data.currentNodeCount,
        location: data.location,
      },
    },
  });
}
