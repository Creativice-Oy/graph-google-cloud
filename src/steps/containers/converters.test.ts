import { createContainerClusterEntity } from './converters';
import { getMockContainerCluster } from '../../../test/mocks';

describe('#createContainerClusterEntity', () => {
  test('should convert to entity', () => {
    expect(
      createContainerClusterEntity(getMockContainerCluster()),
    ).toMatchSnapshot();
  });
});
