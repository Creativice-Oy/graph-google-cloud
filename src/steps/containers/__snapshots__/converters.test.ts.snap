// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createContainerClusterEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Cluster",
  ],
  "_key": "https://container.googleapis.com/v1/projects/j1-gc-integration-dev/locations/us-central1/clusters/j1-gc-integration-dev-gke-cluster",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "addonsConfig": Object {
          "kubernetesDashboard": Object {
            "disabled": true,
          },
          "networkPolicyConfig": Object {
            "disabled": true,
          },
        },
        "autoscaling": Object {},
        "binaryAuthorization": Object {},
        "clusterIpv4Cidr": "10.16.0.0/14",
        "createTime": "2021-02-07T17:17:28+00:00",
        "currentMasterVersion": "1.17.14-gke.1600",
        "currentNodeCount": 3,
        "currentNodeVersion": "1.17.14-gke.1600",
        "databaseEncryption": Object {
          "state": "DECRYPTED",
        },
        "endpoint": "34.69.222.23",
        "initialClusterVersion": "1.17.14-gke.1600",
        "initialNodeCount": 1,
        "instanceGroupUrls": Array [
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-f/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-d0942024-grp",
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-c/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-2a95071a-grp",
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-6817e506-grp",
        ],
        "ipAllocationPolicy": Object {
          "useRoutes": true,
        },
        "labelFingerprint": "a9dc16a7",
        "legacyAbac": Object {},
        "location": "us-central1",
        "locations": Array [
          "us-central1-f",
          "us-central1-c",
          "us-central1-a",
        ],
        "loggingService": "logging.googleapis.com/kubernetes",
        "maintenancePolicy": Object {
          "resourceVersion": "e3b0c442",
        },
        "masterAuth": Object {
          "clusterCaCertificate": "abc123",
        },
        "masterAuthorizedNetworksConfig": Object {},
        "monitoringService": "monitoring.googleapis.com/kubernetes",
        "name": "j1-gc-integration-dev-gke-cluster",
        "network": "default",
        "networkConfig": Object {
          "network": "projects/j1-gc-integration-dev/global/networks/default",
          "subnetwork": "projects/j1-gc-integration-dev/regions/us-central1/subnetworks/default",
        },
        "nodeConfig": Object {
          "diskSizeGb": 100,
          "diskType": "pd-standard",
          "imageType": "COS",
          "machineType": "e2-micro",
          "metadata": Object {
            "disable-legacy-endpoints": "true",
          },
          "oauthScopes": Array [
            "https://www.googleapis.com/auth/cloud-platform",
          ],
          "preemptible": true,
          "serviceAccount": "j1-gc-integration-dev-gke-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
          "shieldedInstanceConfig": Object {
            "enableIntegrityMonitoring": true,
          },
        },
        "nodeIpv4CidrSize": 24,
        "nodePools": Array [
          Object {
            "config": Object {
              "diskSizeGb": 100,
              "diskType": "pd-standard",
              "imageType": "COS",
              "machineType": "e2-micro",
              "metadata": Object {
                "disable-legacy-endpoints": "true",
              },
              "oauthScopes": Array [
                "https://www.googleapis.com/auth/cloud-platform",
              ],
              "preemptible": true,
              "serviceAccount": "j1-gc-integration-dev-gke-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
              "shieldedInstanceConfig": Object {
                "enableIntegrityMonitoring": true,
              },
            },
            "initialNodeCount": 1,
            "instanceGroupUrls": Array [
              "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-f/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-d0942024-grp",
              "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-c/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-2a95071a-grp",
              "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-6817e506-grp",
            ],
            "locations": Array [
              "us-central1-f",
              "us-central1-c",
              "us-central1-a",
            ],
            "management": Object {
              "autoRepair": true,
              "autoUpgrade": true,
            },
            "name": "j1-gc-integration-dev-gke-node-pool",
            "podIpv4CidrSize": 24,
            "selfLink": "https://container.googleapis.com/v1/projects/j1-gc-integration-dev/locations/us-central1/clusters/j1-gc-integration-dev-gke-cluster/nodePools/j1-gc-integration-dev-gke-node-pool",
            "status": "RUNNING",
            "upgradeSettings": Object {
              "maxSurge": 1,
            },
            "version": "1.17.14-gke.1600",
          },
        ],
        "releaseChannel": Object {
          "channel": "REGULAR",
        },
        "selfLink": "https://container.googleapis.com/v1/projects/j1-gc-integration-dev/locations/us-central1/clusters/j1-gc-integration-dev-gke-cluster",
        "servicesIpv4Cidr": "10.19.240.0/20",
        "shieldedNodes": Object {},
        "status": "RUNNING",
        "subnetwork": "default",
        "zone": "us-central1",
      },
    },
  ],
  "_type": "google_container_cluster",
  "active": undefined,
  "clusterIpv4Cidr": "10.16.0.0/14",
  "createdOn": 1612718248000,
  "currentMasterVersion": "1.17.14-gke.1600",
  "currentNodeCount": 3,
  "databaseEncryptionKey": undefined,
  "description": undefined,
  "databaseEncryptionState": "DECRYPTED",
  "displayName": "j1-gc-integration-dev-gke-cluster",
  "endpoint": "34.69.222.23",
  "initialClusterVersion": "1.17.14-gke.1600",
  "initialNodeCode": 1,
  "instanceGroupUrls": Array [
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-f/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-d0942024-grp",
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-c/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-2a95071a-grp",
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instanceGroupManagers/gke-j1-gc-integratio-j1-gc-integratio-6817e506-grp",
  ],
  "kubernetesDashboardEnabled": false,
  "location": "us-central1",
  "locations": Array [
    "us-central1-f",
    "us-central1-c",
    "us-central1-a",
  ],
  "name": "j1-gc-integration-dev-gke-cluster",
  "network": "projects/j1-gc-integration-dev/global/networks/default",
  "networkPolicyConfigEnabled": false,
  "nodeDiskSizeGb": 100,
  "nodeDiskType": "pd-standard",
  "nodeEnableIntegrityMonitoring": true,
  "nodeEnableSecureBoot": undefined,
  "nodeImageType": "COS",
  "nodeIpv4CidrSize": 24,
  "nodeMachineType": "e2-micro",
  "nodeOAuthScopes": Array [
    "https://www.googleapis.com/auth/cloud-platform",
  ],
  "nodePreemptible": true,
  "releaseChannel": "REGULAR",
  "servicesIpv4Cidr": "10.19.240.0/20",
  "status": "RUNNING",
  "subnetwork": "projects/j1-gc-integration-dev/regions/us-central1/subnetworks/default",
  "useIpAliases": false,
  "useRoutes": true,
}
`;
